projects/igniteui-angular/src/lib/query-builder/query-builder.component.spec.ts

```diff
@@ -1984,6 +1984,120 @@ describe('IgxQueryBuilder', () => {
+    });
+
+    describe('API', () => {
+        beforeEach(fakeAsync(() => {
+            queryBuilder.expressionTree = QueryBuilderFunctions.generateExpressionTree();
+            fix.detectChanges();
+            tick(100);
+            fix.detectChanges();
+
+            spyOn(queryBuilder.expressionTreeChange, 'emit').and.callThrough();
+        }));
+
+        it(`Should commit the changes in the edited condition when the 'commit' method is called.`, fakeAsync(() => {
+            // Double-click the existing chip to enter edit mode.
+            QueryBuilderFunctions.clickQueryBuilderTreeExpressionChip(fix, [1], true);
+            tick(50);
+            fix.detectChanges();
+
+            // Change the current condition
+            QueryBuilderFunctions.selectColumnInEditModeExpression(fix, 1);
+            expect(queryBuilder.canCommit()).toBeFalse();
+            QueryBuilderFunctions.selectOperatorInEditModeExpression(fix, 0);
+            expect(queryBuilder.canCommit()).toBeFalse();
+            const input = QueryBuilderFunctions.getQueryBuilderValueInput(fix).querySelector('input');
+            UIInteractions.clickAndSendInputElementValue(input, 'a');
+            tick(100);
+            fix.detectChanges();
+
+            // Apply the changes
+            queryBuilder.commit();
+            tick(100);
+            fix.detectChanges();
+
+            // Verify expression is commited
+            QueryBuilderFunctions.verifyExpressionChipContent(fix, [1], 'OrderName', 'Contains', 'a');
+
+            // Verify event is not fired
+            expect(queryBuilder.expressionTreeChange.emit).toHaveBeenCalledTimes(0);
+        }));
+
+        it(`Should discard the changes in the edited condition when the 'discard' method is called.`, fakeAsync(() => {
+            // Double-click the existing chip to enter edit mode.
+            QueryBuilderFunctions.clickQueryBuilderTreeExpressionChip(fix, [1], true);
+            tick(50);
+            fix.detectChanges();
+
+            // Change the current condition
+            QueryBuilderFunctions.selectColumnInEditModeExpression(fix, 1);
+            expect(queryBuilder.canCommit()).toBeFalse();
+            QueryBuilderFunctions.selectOperatorInEditModeExpression(fix, 0);
+            expect(queryBuilder.canCommit()).toBeFalse();
+            const input = QueryBuilderFunctions.getQueryBuilderValueInput(fix).querySelector('input');
+            UIInteractions.clickAndSendInputElementValue(input, 'a');
+            tick(100);
+            fix.detectChanges();
+
+            // Apply the changes
+            queryBuilder.discard();
+            tick(100);
+            fix.detectChanges();
+
+            // Verify expression is not commited
+            QueryBuilderFunctions.verifyExpressionChipContent(fix, [1], 'OrderId', 'Greater Than', '3');
+
+            // Verify event is not fired
+            expect(queryBuilder.expressionTreeChange.emit).toHaveBeenCalledTimes(0);
+        }));
```

Comment:
Please add a couple tests covering the scenario where we actually commit an invalid condition and check it doesn't throw.